pkg/clients/aws/route53.go

```diff
@@ -123,13 +126,23 @@ func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken
 		return "", err
 	}
 
-	baseDomain := cr.Spec.ACMEDNSDomain
-	if !strings.HasSuffix(baseDomain, ".") {
-		baseDomain = baseDomain + "."
+	dnsZones := hivev1.DNSZoneList{}
+	err = c.kubeClient.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		return "", err
+	}
+
+	if len(dnsZones.Items) == 0 {
+		return "", fmt.Errorf("no dns zones found")
```

Comment:
Can this error be a bit more descriptive? If I was to get this error message I would be thinking the route53 zone was missing, but that's not what it's checking. It's checking if there are and DNSZone objects in a specific namespace.